data-processing-lib/src/data_processing/pure_python/transform_table_processor.py

```diff
@@ -0,0 +1,190 @@
+#  http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import time
+import traceback
+from typing import Any
+
+import pyarrow as pa
+from data_processing.data_access import DataAccessFactoryBase
+from data_processing.transform import TransformConfiguration, TransformStatistics
+from data_processing.utils import TransformUtils, get_logger
+
+
+logger = get_logger(__name__)
+
+
+class TransformTableProcessor:
+    """
+    This is the class implementing the worker class processing of a single pyarrow file
+    """
+
+    def __init__(
+        self,
+        data_access_factory: DataAccessFactoryBase,
+        statistics: TransformStatistics,
+        params: TransformConfiguration,
+    ):
+        """
+        Init method
+        :param data_access_factory - data access factory
+        :param statistics - reference to statistics class
+        :param params: transform configuration class
+        """
+        # Create data access
+        self.data_access = data_access_factory.create_data_access()
+        # Add data access and statistics to the processor parameters
+        transform_params = dict(params.get_transform_params())
+        transform_params["data_access"] = self.data_access
+        transform_params["statistics"] = statistics
+        # Create local processor
+        self.transform = params.get_transform_class()(transform_params)
+        # Create statistics
+        self.stats = statistics
+        self.last_fname = None
+        self.last_fname_next_index = None
+
+    def process_data(self, f_name: str) -> None:
+        """
+        Method processing an individual file
+        :param f_name: file name
+        :return: None
+        """
+        logger.debug(f"Begin processing file {f_name}")
+        if self.data_access is None:
+            logger.warning("No data_access found. Returning.")
+            return
+        t_start = time.time()
+        # Read source table
+        table = self.data_access.get_table(path=f_name)
+        if table is None:
+            logger.warning("File read resulted in None. Returning.")
+            self.stats.add_stats({"failed_reads": 1})
+            return
+        self.stats.add_stats({"source_files": 1, "source_size": table.nbytes})
+        # Process input table
+        try:
+            if table.num_rows > 0:
+                # execute local processing
+                logger.debug(f"Begin transforming table from {f_name}")
+                out_tables, stats = self.transform.transform(table=table)
+                logger.debug(f"Done transforming table from {f_name}")
```

Comment:
we can add here the length of the out_tables

Reply:
nah